### PR TITLE
Call PlayerLaunchProjectileEvent for wind charge

### PR DESCRIPTION
**Pull request by Lulu13022002**

### CODE REVIEW

File: patches/server/0215-PlayerLaunchProjectileEvent.patch [hunk 1]
```diff
@@ -317,3 +317,47 @@ index 2d34f206b4b02d5f2ee35101233afc6b4f58a579..f1b2d388a1a40a1d909a2e726f32d6c1
...
                                  // CraftBukkit start - SPIGOT-5458 also need in this branch :(
+diff --git a/src/main/java/net/minecraft/world/item/WindChargeItem.java b/src/main/java/net/minecraft/world/item/WindChargeItem.java
+index 6222e867bb959fab05bcd8f2114ab00b26847c3e..bdf0f9177a65e70147311e71784c796f7edaf2af 100644
+--- a/src/main/java/net/minecraft/world/item/WindChargeItem.java
++++ b/src/main/java/net/minecraft/world/item/WindChargeItem.java
+@@ -24,10 +24,26 @@ public class WindChargeItem extends Item implements ProjectileItem {
+ 
+     @Override
+     public InteractionResultHolder<ItemStack> use(Level world, Player user, InteractionHand hand) {
++        ItemStack itemStack = user.getItemInHand(hand); // Paper - moved from below
```
Comment (lynxplay) [inline patches/server/0215-PlayerLaunchProjectileEvent.patch]: // Paper - PlayerLaunchProjectileEvent - moved from below.

File: patches/server/0215-PlayerLaunchProjectileEvent.patch [hunk 2]
```diff
@@ -317,3 +317,47 @@ index 2d34f206b4b02d5f2ee35101233afc6b4f58a579..f1b2d388a1a40a1d909a2e726f32d6c1
...
+         if (!world.isClientSide()) {
+             WindCharge windCharge = new WindCharge(user, world, user.position().x(), user.getEyePosition().y(), user.position().z());
+             windCharge.shootFromRotation(user, user.getXRot(), user.getYRot(), 0.0F, 1.5F, 1.0F);
+-            world.addFreshEntity(windCharge);
++            // Paper start - PlayerLaunchProjectileEvent
++            com.destroystokyo.paper.event.player.PlayerLaunchProjectileEvent event = new com.destroystokyo.paper.event.player.PlayerLaunchProjectileEvent((org.bukkit.entity.Player) user.getBukkitEntity(), org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(itemStack), (org.bukkit.entity.Projectile) windCharge.getBukkitEntity());
++            if (!event.callEvent() || !world.addFreshEntity(windCharge)) {
++                user.containerMenu.sendAllDataToRemote();
++                if (user instanceof net.minecraft.server.level.ServerPlayer player) {
++                    player.connection.send(new net.minecraft.network.protocol.game.ClientboundCooldownPacket(this, 0));
```
Comment (lynxplay) [inline patches/server/0215-PlayerLaunchProjectileEvent.patch]: Maybe worth some comment as to why this is needed?
given that iirc this is the first time we manually sent it in the codebase.

**Review by lynxplay — COMMENTED** [review]
not tested but diff looks good 👍 will test this weekend.

File: patches/server/0215-PlayerLaunchProjectileEvent.patch (hunk 2 above)
Comment (Lulu13022002) [reply patches/server/0215-PlayerLaunchProjectileEvent.patch]: Yes similar thing should be done for the enderpearl too

**Review by lynxplay — APPROVED** [review]

**Review by electronicboy — APPROVED** [review]

**Pull request merged**